samcli/lib/telemetry/metric.py

```diff
@@ -138,14 +140,21 @@ def wrapped(*args, **kwargs):
 
         try:
             ctx = Context.get_current_context()
+            try:
+                template_dict = ctx.template_dict
+            except AttributeError:
+                LOG.debug("Ignoring CDK project check as template is not provided in context.")
+                template_dict = {}
+            project_type = ProjectTypes.CDK.value if is_cdk_project(template_dict) else ProjectTypes.CFN.value
```

Comment:
What if a user run a command at the location where there is neither CFN nor CDK project?

Reply:
Similar to Mohamed's comment, if the template is not provided, we don't add the project type metric. For now, we only have these two types of templates, so if template is provided, it can be either CFN, or CDK.